components/docs-chef-io/content/automate/major_upgrade_4.x.md

```diff
@@ -0,0 +1,315 @@
+- **Current Version should be 3.0.x** If you are not on this version, please do normal upgrade as per your topology.
+
+## Upgrade to version 3.0.x
+
+Check your current version:
+```sh
+sudo chef-automate version
+```
+If your Server Version is less than 20220329091442, follow the link to [upgrade to 3.0.x]({{< relref "major_upgrade.md" >}})
+
+If your Server Version is less than 3.0.x.
+Please upgrade to latest minor version in 3 series.
+### Airgapped upgrade to 3.0.x
+- On Internet connected machine
+  - Download latest chef-automate cli.
+  ```sh
+  curl https://packages.chef.io/files/current/latest/chef-automate-cli/chef-automate_linux_amd64.zip | gunzip - > chef-automate && chmod +x chef-automate
+  ```
+  - Create bundle of version 3.0.x
+  ```sh
+  ./chef-automate airgap bundle create --version 3.0.x
+  ```
+  - Copy this bundle file automate-3.0.x.aib and latest downloaded cli `chef-automate` to Automate Machine in airgap.
+- On Automate Machine in Airgap
+  - Upgrade Automate with bundle
+  ```sh
+  sudo ./chef-automate upgrade run --airgap-bundle automate-3.0.x.aib
+  ```
+  - Check all services are up and running
+  ```sh
+  sudo chef-automate status
+  ```
+### Normal Upgrade to 3.0.x
+- Upgrade Chef Automate to latest minor version (3.0.x)
+
+  ```sh
+  sudo chef-automate upgrade run
+  ```
+- Check all services are up and running
+
+  ```sh
+  sudo chef-automate status
+  ```
+
+
+## Upgrade Path to version 4.0.x from 3.0.x
+
+There are four possible upgrade scenarios:
+
+- [Chef Automate with Embedded Elasticsearch]({{< relref "#chef-automate-with-embedded-elasticsearch" >}})
+- [Chef Automate with External Elasticsearch]({{< relref "#chef-automate-with-external-elasticsearch" >}})
+- [Chef Automate in Air-Gapped Environment With Embedded Elasticsearch]({{< relref "#chef-automate-in-air-gapped-environment-with-embedded-elasticsearch" >}})
+- [Chef Automate in Air-Gapped Environment With External Elasticsearch]({{< relref "#chef-automate-in-air-gapped-environment-with-external-elasticsearch" >}})
+
+{{< note >}}
+
+If you are unsure if your installation of Chef Automate is using an external elasticsearch, run `chef-automate config show`. You are using an external elasticsearch if `enable=true` is present in the `global.v1.external.elasticsearch` config setting.
+
+{{< /note >}}
+
+{{< warning >}}
+
+Sixty percent of your drive should be free space before starting a major version upgrade.
+
+{{< /warning >}}
+
+{{< warning >}}
+
+Sharding needs to be disabled for automate running embedded elasticsearch. Please accept the checklist item asking permission to disable sharding.
+
+{{< /warning >}}
+
+### Chef Automate With Embedded Elasticsearch
+
+To upgrade Chef Automate with embedded Elasticsearch, follow the steps given below:
+
+**Upgrade Chef Automate from version 3.0.x to 4.0.x**
+
+1. Start a major version upgrade:\
```

Comment:
how about mentioning the pre checklists as mandatory steps to do before the upgrade in the doc, so they will perform before their upgrade starts.

Reply:
its given under prerequisites